backend/app/services/answer_generator.py

```
@@ -22,24 +22,40 @@ def __init__(self):
         # Prioritize OpenAI
-        if provider == "openai" or (not provider and settings.OPENAI_API_KEY):
+        if provider == "openai" or (not provider and api_key):
             try:
                 from langchain_openai import ChatOpenAI
                 self.llm = ChatOpenAI(
-                    model=settings.NL_DEFAULT_MODEL or "gpt-4o-mini",
+                    model=model or "gpt-4o-mini",
+                    api_key=api_key,
                     temperature=0.3,  # Slightly creative but still factual
                 )
             except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)